DAO.sol

```diff
@@ -490,14 +490,24 @@ contract DAO is DAOInterface, Token, TokenSale {
             throw;
         }
 
-        if (p.newServiceProvider) {
-            p.open = false;
+        if (p.newCurators) {
+            if (now > p.votingDeadline + 28) {
+                p.open = false;
+            }
             return;
         }
 
         if (p.amount > actualBalance())
             throw;
 
+        if (!allowedRecipients[p.recipient]) {
```

Comment:
Add a comment above here explaining the extraneous conditions when this will happen. Example:
```
// If the curators removed the recipient from the whitelist, close the proposal
// in order to free the deposit and allow unblocking of voters
```